config.go

```diff
@@ -0,0 +1,60 @@
+package carrot
+
+import (
+	"gopkg.in/yaml.v2"
+	"sync"
+	"io/ioutil"
+	"log"
+	"time"
+)
+
+var (
+	once     sync.Once
+	instance Config
+)
+
+type Config struct {
+	Server struct {
+		BroadcastChannelSize int    `yaml:"broadcast_channel_size"`
+		Port                 int    `yaml:"port"`
+		ServerSecret         string `yaml:"server_secret"`
+	}
+	Session struct {
+		NilSessionToken                     SessionToken  `yaml:"nil_session_token"`
+		DefaultSessionClosedTimeoutDuration time.Duration `yaml:"default_sess_closed_timeout_duration_secs"`
+	}
+	Router struct {
```

Comment:
There should be router settings and dispatcher settings. They two different modules